plugins/VertexTableProducer.cc

```diff
@@ -116,14 +120,24 @@ void
 VertexTableProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 {
     using namespace edm;
+    edm::Handle<edm::ValueMap<float>> pvsScoreIn;
     edm::Handle<std::vector<reco::Vertex>> pvsIn;
     iEvent.getByToken(pvs_, pvsIn);
-    auto pvsTable = std::make_unique<FlatTable>(1,pvName_,true);
-    pvsTable->addColumnValue<float>("ndof",(*pvsIn)[0].ndof(),"",FlatTable::FloatColumn);
-    pvsTable->addColumnValue<float>("x",(*pvsIn)[0].position().x(),"vertex position x coordinate",FlatTable::FloatColumn);
-    pvsTable->addColumnValue<float>("y",(*pvsIn)[0].position().y(),"vertex position y coordinate",FlatTable::FloatColumn);
-    pvsTable->addColumnValue<float>("z",(*pvsIn)[0].position().z(),"vertex position z coordinate",FlatTable::FloatColumn);
-    pvsTable->addColumnValue<float>("chi2",(*pvsIn)[0].normalizedChi2(),"reduced chi2",FlatTable::FloatColumn);
+    iEvent.getByToken(pvsScore_, pvsScoreIn);
+    auto pvTable = std::make_unique<FlatTable>(1,pvName_,true);
+    pvTable->addColumnValue<float>("ndof",(*pvsIn)[0].ndof(),"main primary vertex number of degree of freedom",FlatTable::FloatColumn);
+    pvTable->addColumnValue<float>("x",(*pvsIn)[0].position().x(),"main primary vertex position x coordinate",FlatTable::FloatColumn);
+    pvTable->addColumnValue<float>("y",(*pvsIn)[0].position().y(),"main primary vertex position y coordinate",FlatTable::FloatColumn);
+    pvTable->addColumnValue<float>("z",(*pvsIn)[0].position().z(),"main primary vertex position z coordinate",FlatTable::FloatColumn);
+    pvTable->addColumnValue<float>("chi2",(*pvsIn)[0].normalizedChi2(),"main primary vertex reduced chi2",FlatTable::FloatColumn);
+    pvTable->addColumnValue<int>("npvs",(*pvsIn).size(),"total number of reconstructed primary vertices",FlatTable::IntColumn);
+    pvTable->addColumnValue<float>("score",(*pvsScoreIn).get(pvsIn.id(),0),"main primary vertex score, i.e. sum pt2 of clustered objects",FlatTable::FloatColumn,8);
+
+    auto otherPVsTable = std::make_unique<FlatTable>((*pvsIn).size() >3?3:(*pvsIn).size() ,"Other"+pvName_,false);
+    std::vector<float> pvsz;
+    for(size_t i=0;i < (*pvsIn).size() && i < 3; i++) pvsz.push_back((*pvsIn)[0].position().z());
```

Comment:
`[i]` instead of `[0]`

Reply:
yeah that's seems a better idea, plus I should start from 1 given that the first is already stored... fixing it